File: frontend/cinemas.js
// Загрузка списка кинотеатров
function loadCinemas() {
    fetch("http://localhost:8080/api/cinemas")
        .then(response => response.json())
        .then(data => {
            const tbody = document.querySelector("#cinemas-table tbody");
            tbody.innerHTML = "";
            data.forEach(cinema => {
                const row = document.createElement("tr");
                row.innerHTML = `
                    <td>${cinema.name}</td>
                    <td>${cinema.address}</td>
                    <td>${cinema.category}</td>
                    <td>${cinema.halls}</td>
                    <td>${cinema.seats}</td>
                    <td>${cinema.status}</td>
                `;
                tbody.appendChild(row);
            });
        })
        .catch(() => {
            alert("Не удалось загрузить список кинотеатров.");
        });
}

// Отправка нового кинотеатра
function createCinema(event) {
    event.preventDefault();

    const cinema = {
        name: document.getElementById("cinema-name").value,
        address: document.getElementById("cinema-address").value,
        category: document.getElementById("cinema-category").value,
        halls: parseInt(document.getElementById("cinema-halls").value),
        seats: parseInt(document.getElementById("cinema-seats").value),
        status: document.getElementById("cinema-status").value
    };

    fetch("http://localhost:8080/api/cinemas", {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify(cinema)
    })
        .then(response => {
            if (!response.ok) throw new Error();
            alert("Кинотеатр успешно добавлен!");
            document.querySelector("form").reset();
            loadCinemas();
        })
        .catch(() => {
            alert("Ошибка при добавлении кинотеатра.");
        });
}

File: back/handlers/cinemas.go
package handlers

import (
	"encoding/json"
	"net/http"

	"cinema.local/db"
	"cinema.local/models"
)

// GET /api/cinemas
func GetCinemas(w http.ResponseWriter, r *http.Request) {
	rows, err := db.GetDB().Query(`SELECT id, name, address, category, halls, seats, status FROM cinemas`)
	if err != nil {
		http.Error(w, "Ошибка при получении списка кинотеатров", http.StatusInternalServerError)
		return
	}
	defer rows.Close()

	var cinemas []models.Cinema
	for rows.Next() {
		var c models.Cinema
		err := rows.Scan(&c.ID, &c.Name, &c.Address, &c.Category, &c.Halls, &c.Seats, &c.Status)
		if err != nil {
			http.Error(w, "Ошибка при чтении строки", http.StatusInternalServerError)
			return
		}
		cinemas = append(cinemas, c)
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(cinemas)
}

// POST /api/cinemas
func CreateCinema(w http.ResponseWriter, r *http.Request) {
	var c models.Cinema
	if err := json.NewDecoder(r.Body).Decode(&c); err != nil {
		http.Error(w, "Неверный формат данных", http.StatusBadRequest)
		return
	}

	_, err := db.GetDB().Exec(
		`INSERT INTO cinemas (name, address, category, halls, seats, status) VALUES ($1, $2, $3, $4, $5, $6)`,
		c.Name, c.Address, c.Category, c.Halls, c.Seats, c.Status,
	)
	if err != nil {
		http.Error(w, "Ошибка при добавлении кинотеатра", http.StatusInternalServerError)
		return
	}

	w.WriteHeader(http.StatusCreated)
}
